core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -794,7 +794,15 @@ final class ShuffleBlockFetcherIterator(
             // since the last call.
             val msg = s"Received a zero-size buffer for block $blockId from $address " +
               s"(expectedApproxSize = $size, isNetworkReqDone=$isNetworkReqDone)"
-            throwFetchFailedException(blockId, mapIndex, address, new IOException(msg))
+            if (blockId.isShuffleChunk) {
+              logWarning(msg)
+              pushBasedFetchHelper.initiateFallbackFetchForPushMergedBlock(blockId, address)
```

Comment:
What is the size that is returned in the meta request?

Reply:
did you mean PushMergedRemoteMetaFetchResult? 
The size of push-merged block is not zero, since the size of each chunk cannot be obtained on the reduce side, we print the zero-size log in the following code on the server side, and confirm that the indexFile has the same offset continuously. https://github.com/apache/spark/blob/9a7596e1dde0f1dd596aa6d3b2efbcb5d1ef70ea/core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala#L500

Then according to the hardware layer error information, we basically determine that the problem of data loss occurs in the process of writing data.

Reply:
And then the size of the original shuffle block is not 0, because there is no FetchFailedException in the stage

Reply:
So we can read the meta and index files without any issues but when we read the data, there are problems and size of the chunk is 0. Yeah I think the change looks good to me.

Reply:
Can you update the comment `initiateFallbackFetchForPushMergedBlock` with this scenario as well.

Reply:
ok, updated